In [1]:
import numpy as np
import pandas as pd


In [2]:
df= pd.read_csv("C:/Sharayu Portfolio/Sharayu Project/Vestiaire Fashion Data/vestiaire.csv/vestiaire.csv")
df.head(3)

product_id     product_type  \
0    43247626  Wool mini skirt   
1    43247441           Jacket   
2    43246517        Wool coat   

                                        product_name  \
0  Wool mini skirt Miu Miu Grey size S Internatio...   
1      Jacket Barbara Bui Navy size 42 FR in Cotton    
2  Wool coat Comme Des Garcons White size S Inter...   

                                 product_description  \
0  Miu Miu – Pleated mini skirt Size: 36 (S)  Wai...   
1  For selling nice women's suit Barbara Bui size...   
2  Magnificent boiled wool coat. I bought it in t...   

               product_keywords product_gender_target product_category  \
0           Miu Miu Wool Skirts                 Women   Women Clothing   
1    Barbara Bui Cotton Jackets                 Women   Women Clothing   
2  Comme Des Garcons Wool Coats                 Women   Women Clothing   

    product_season    product_condition  product_like_count  ...  \
0  Autumn / Winter           Never worn                34.0  ...   
1      All seasons  Very good condition                 1.0  ...   
2  Autumn / Winter  Very good condition                 2.0  ...   

   warehouse_name  seller_id  seller_username  usually_ships_within  \
0       Tourcoing   25775970  vitalii25775970                   NaN   
1       Tourcoing   13698770   olivia13698770                   NaN   
2       Tourcoing    6042365   cecilia6042365              1-2 days   

   seller_country  seller_products_sold seller_num_products_listed  \
0         Germany                   3.0                       14.0   
1         Belgium                   0.0                        0.0   
2           Spain                  58.0                       69.0   

  seller_community_rank seller_num_followers seller_pass_rate  
0                   0.0                 13.0              0.0  
1                   0.0                  8.0              0.0  
2                   0.0                 62.0             96.0  

[3 rows x 36 columns]

In [3]:
df.shape

(900514, 36)

In [4]:
# Count total rows
total_rows = len(df)

# Count rows with at least one null value
rows_with_null = df[df.isnull().any(axis=1)].shape[0]

rows_with_null


168295

In [5]:
# Calculate percentage of data that would be lost
percentage_lost = (rows_with_null / total_rows) * 100

print(f"Percentage of data that would be lost: {percentage_lost:.2f}%")

Percentage of data that would be lost: 18.69%


### If we drop the null values. We will end of losing 18.69% of the data. As this is a significant amount, we will consider performing imputations for the missing values. 

### Lets calculate the percentage of null values for each column to check which columns to perform imputations on

In [6]:
# Calculate the percentage of null values for each column
null_percentage = df.isnull().mean() * 100

# Display the percentages
print(null_percentage)

product_id                     0.000000
product_type                   0.000000
product_name                   0.000000
product_description            0.000777
product_keywords               0.131369
product_gender_target          0.000000
product_category               0.131369
product_season                 0.000222
product_condition              0.000000
product_like_count             0.000000
sold                           0.000000
reserved                       0.000000
available                      0.000000
in_stock                       0.000000
should_be_gone                 0.000000
brand_id                       0.000000
brand_name                     0.000000
brand_url                      0.000000
product_material               0.000444
product_color                  0.000111
price_usd                      0.000000
seller_price                   0.000000
seller_earning                 0.000000
seller_badge                   0.000000
has_cross_border_fees          1.525351


### We will perform imputations on the following columns: usually_ships_within , product_category, product_keywords, buyers_fees and has_cross_border_fees 

### For the categorical columns we will fill in missing values using mode and for buyer_fees we will use median

In [7]:
df['usually_ships_within'] = df['usually_ships_within'].fillna(df['usually_ships_within'].mode()[0])
df['product_category'] = df['product_category'].fillna(df['product_category'].mode()[0])
df['product_keywords'] = df['product_keywords'].fillna(df['product_keywords'].mode()[0])

In [8]:
df['buyers_fees'] = df['buyers_fees'].fillna(df['buyers_fees'].median())

### To perform imputation on column 'has_cross_border_fees' we will first check the values in this column

In [9]:
df[['has_cross_border_fees']].value_counts()

has_cross_border_fees
True                     886778
dtype: int64

### Because all existing values are True. Therefore, for any missing values in this column, we will replace them with 'False'.

In [10]:
df['has_cross_border_fees'].fillna(False, inplace=True)

### Verify the imputation

In [11]:
columns_imputed = ['usually_ships_within', 'product_category', 'product_keywords', 'buyers_fees', 'has_cross_border_fees']

# Check null values in these columns
for column in columns_imputed:
    null_count = df[column].isnull().sum()
    print(f"{column}: {null_count} null values")

usually_ships_within: 0 null values
product_category: 0 null values
product_keywords: 0 null values
buyers_fees: 0 null values
has_cross_border_fees: 0 null values


In [12]:
print(f'Percentage of missing values: {(df.isna().mean() * 100).sum().round(4)}')

Percentage of missing values: 0.0059


### Now we can drop the remaining null values

In [13]:
df.dropna(inplace=True)

In [14]:
print(f'Percentage of missing values: {(df.isna().mean() * 100).sum()}')

Percentage of missing values: 0.0


### Now our dataset is free of null values

In [15]:
df.shape

(900461, 36)

### Check for duplicates

In [16]:
print(df.duplicated().sum())

0


### There are not any duplicate values. Now we can use this dataset for our analysis

### Download the cleaned dataset

In [20]:
# Specify the path where you want to save the file
file_path = 'C:\Sharayu Portfolio\Sharayu Project\Vestiaire Fashion Data\Vers Clean Data\Vers Cleaned Data.csv'

# Save the DataFrame as a CSV file
df.to_csv(file_path, index=False)

print(f"Dataset has been saved as '{file_path}'")


Dataset has been saved as 'C:\Sharayu Portfolio\Sharayu Project\Vestiaire Fashion Data\Vers Clean Data\Vers Cleaned Data.csv'
